File: C++/TowerOfHanoi.cpp
#include <bits/stdc++.h>
using namespace std;
void TOH(char source, char destination, char helper, int n)
{
    if (n == 0)
        return;
    TOH(source, helper, destination, n - 1);
    cout << "Move " << source << " to " << destination << endl;
    TOH(helper, destination, source, n - 1);
}

int main(int argc, char const *argv[])
{
    int n;
    cout << "Enter number of tiles : " << endl;
    cin >> n;
    if (n == 0)
    {
        cout << "No Tiles.";
    }
    else if (n == 1)
    {
        cout << "Move A to B." << endl;
    }
    else
    {
        TOH('A', 'C', 'B', n);
    }
    return 0;
}

File: Data Structures and Algorithms in Java/Recursion/Tower_Of_Hanoi.java
class Tower_Of_Hanoi{
    public static void main(String[] args) {
        toh(3,'A','B','C');
    }
    static void toh(int n,char A,char B,char C){
        if(n==1){
            System.out.println("Move 1 from Tower "+A+ " to Tower "+C);
            return;
        }
        toh(n-1,A,C,B);
        System.out.println("Move "+n+ " from Tower "+A+" to Tower "+C);
        toh(n-1,B,A,C);
    }
}